presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/CanonicalizeExpressionRewriter.java

```diff
@@ -42,6 +45,38 @@ public static Expression canonicalizeExpression(Expression expression)
     private static class Visitor
             extends ExpressionRewriter<Void>
     {
+        @Override
+        public Expression rewriteComparisonExpression(ComparisonExpression node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
+        {
+            // if we have a comparison of the form <constant> <op> <expr>, normalize it to
+            // <expr> <op-flipped> <constant>
+            if (node.getLeft() instanceof Literal && !(node.getRight() instanceof Literal)) {
+                return new ComparisonExpression(
+                        node.getOperator().flip(),
+                        treeRewriter.rewrite(node.getRight(), context),
+                        treeRewriter.rewrite(node.getLeft(), context));
+            }
+
+            return treeRewriter.defaultRewrite(node, context);
+        }
+
+        @Override
+        public Expression rewriteArithmeticBinary(ArithmeticBinaryExpression node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
```

Comment:
How it would work for: `((a + 1) + b)`, shouldn't it rewrite that to `(a+b) + 1`

Reply:
+1, you could recursively extract subtree of additions, multiplications and work on operand lists instead of just two.
I think we do something similar in `ExtractCommonPredicatesExpressionRewriter` for `AND` and `ORs`

Reply:
That's a good idea, but out of scope for this PR :)